gherkin8/src/main/java/io/cucumber/core/gherkin5/Gherkin8CucumberFeatureParser.java

```diff
@@ -0,0 +1,62 @@
+package io.cucumber.core.gherkin5;
+
+import io.cucumber.core.gherkin.CucumberFeature;
+import io.cucumber.core.gherkin.CucumberFeatureParser;
+import io.cucumber.core.gherkin.CucumberParserException;
+import io.cucumber.core.gherkin.CucumberPickle;
+import io.cucumber.gherkin.Gherkin;
+import io.cucumber.gherkin.GherkinDialect;
+import io.cucumber.gherkin.GherkinDialectProvider;
+import io.cucumber.gherkin.GherkinDocumentBuilder;
+import io.cucumber.gherkin.Parser;
+import io.cucumber.gherkin.ParserException;
+import io.cucumber.gherkin.pickles.PickleCompiler;
+import io.cucumber.messages.Messages;
+
+import java.io.ByteArrayInputStream;
+import java.net.URI;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class Gherkin8CucumberFeatureParser implements CucumberFeatureParser {
+    private static CucumberFeature parseGherkin5(URI path, String source) {
+        try {
+
+            Stream<Messages.Envelope> messages = Gherkin.fromStream(new ByteArrayInputStream(source.getBytes(UTF_8)));
+
+            Parser<Messages.GherkinDocument.Builder> parser = new Parser<>(new GherkinDocumentBuilder());
+            Messages.GherkinDocument gherkinDocument = parser.parse(source).setUri(path.toString()).build();
```

Comment:
Not sure where this URI goes but if it comes out again as string it might be useful. Would be better if this was a string.